internal/services/v1/relationships.go

```diff
@@ -290,13 +325,18 @@ func (ps *permissionServer) DeleteRelationships(ctx context.Context, req *v1.Del
 			return err
 		}
 
+		if len(deleteMutations) > 0 {
+			return rwt.WriteRelationships(ctx, deleteMutations)
+		}
```

Comment:
This strategy is fine for now, but it requires to read all relationships and then issue write. Why wasn't `DELETE ... LIMIT` considered? If we used the relationship filter, it would provide a stable order to delete for. Was it because it wouldn't be possible to know if there are further rows to delete?

I also wondered if all the datastores we use are configured to run with serializable isolation level (I recall some proposals to relax PG). I think this would be relevant because it could be subject to phantom reads problem - you read 1000 rels, but some other transaction deleted 3 rels of your same set. Effectively when `WriteRelationships` is run, you've deleted 997 rels, thus not honored the limit

Reply:
`DELETE ... LIMIT` is not supported on some datastores (such as MySQL)

This is all running under a transaction, so it should be safe, count wise